heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java

```diff
@@ -374,13 +389,12 @@ public boolean submitTopology() {
 
       // create an instance of uploader
       uploader = ReflectionUtils.newInstance(uploaderClass);
+
     } catch (IllegalAccessException | InstantiationException | ClassNotFoundException e) {
-      LOG.log(Level.SEVERE, "Failed to instantiate instances", e);
-      return false;
+      throw new TopologySubmissionException(
+          String.format("Failed to instantiate instances: %s", e.getMessage()), e);
```

Comment:
e.getMessage() is included in the exception chain when you chain the message, so you would typically not re-included it in a custom message here. Better to replace e.getMessage() with uploaderClass actually.